wunderio/lando-drupal
# Feature/20 add extensions, change install steps

## Description

Currently in https://github.com/wunderio/drupal-project/blob/main/.lando.yml and this project our Lando yml file contains lot of services. Some are also commented out in case you need them. In one of our projects, we don't need **Node** so how to make that optional? There should be an easy way to pick what you need and what not.

This feature adds a custom extension system and adds the first **node** extension. It's solved by using the [yq](https://github.com/mikefarah/yq) linux tool to merge 2 yml files together. 

The system does not yet have functionality to also add custom scripts, but that can be added later.

This PR also removes requirement for [koodimonni/composer-dropin-installer](https://github.com/Koodimonni/Composer-Dropin-Installer) as I realized how the dropin plugin was created and actually borrowed some code from it.

In addition to that we'll be changing the general install logic of requiring to add both the .lando.base.yml and .lando/core to each project's own repo. This way:

1. We see what changes are coming in when requiring new version of wunderio/lando-drupal
2. On fresh install of project, lando start would work. There are no hooks to fetch all of the code before .lando.yml file is red in. Everything has to be there already after cloning.

## Todo?

2. How to uninstall?
Add `unload-wunderio-lando-drupal-extensions`?

## Testing

This test is based on the guess that you don't yet have **wunderio/lando-drupal** installed.

1. Remove node service and npm tooling from your .lando.yml

4. Install current PR's  **wunderio/lando-drupal** composer package:

`lando composer require wunderio/lando-drupal:dev-feature/20-add-extensions --dev`

5. Rebuild lando:

`lando rebuild`

You are now using **wunderio/lando-drupal** package minus the Node service.

7. Load the node extension into .lando.yml and .lando.base.yml with the new tooling command:

`lando load-wunderio-lando-drupal-extensions node`

It will first write this into your local .lando.yml:

```
wunderio:
  extensions:
    - node
```

Then it'll merge the node changes into .lando.base.yml. Check that you can see the Node service and npm tooling in .lando.base.yml. Basically this https://github.com/wunderio/lando-drupal/pull/21/files#diff-2295bf9cd69f06ec8e2b148811e61665089a0a1354de813024b9d02de01e2f29 was merged into the .lando.base.yml

8. Rebuild Lando:

```
   lando rebuild
```

You should have Node service available together with `lando npm` tooling.
